docs/doc_gen.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python
+#
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generates user_guide.md from user_guide.md.in ."""
+
+import json
+import os
+import re
+import sys
+
+
+def ReadFileRaw(filename):
+    with open(filename, 'r') as source:
+        for line in source:
+            yield line
+
+
+def ReadFileContentsWithMarker(filename, marker):
```

Comment:
It would be nice to add a function comment here that describes what this is doing.  Same for the other functions in this file along with a top level comment that describes the intent of this script.

Reply:
Done.